### ISSUE TITLE
Analytics: Remove Guest Customer Data from Customers Table Upon Order Delete

### ISSUE THREAD

**Issue by timmyc** [woocommerce/woocommerce-admin#4946]
While discussing data retention with various folks in #3810 - it was noted that we are not adhering to the Personal Data Retention Settings that WooCommerce offers:

<img width="906" alt="image" src="https://user-images.githubusercontent.com/22080/89558561-3bce5e00-d7c9-11ea-83c5-122e787b1e6d.png">

Specifically the "Retain Completed Orders" setting.

The proposed solution is:

- When an order is deleted, regardless of reason, either via Retain Completed Orders period lapsing - or by the order object itself being deleted by an admin user, any related data in the Customers Reporting Table should be also deleted unless:
- If *another* order exists for the given `user_id` or guest checkout email address, the the customer reporting table should remain.

**Comment by timmyc** [comment]
I think the estimate of 3 is still a bit high, but feels like a safe one for this task.

**Comment by adrianduffell** [comment]
I've performed some investigation here.

Notably the `Personal data retention` settings don't cause orders to be deleted. Instead, the process "anonymizes" the order by scrubbing personal information. This is how an anonymized order looks in WooComerce:

![Screen Shot 2020-09-23 at 10 22 33 pm](https://user-images.githubusercontent.com/9312929/94025648-51d0c600-fdeb-11ea-94bd-3a0ddef7da5c.png)

Likewise, the WC Admin analytics data for the order is also anonymized during this process:

![Screen Shot 2020-09-23 at 10 24 11 pm](https://user-images.githubusercontent.com/9312929/94025850-8ba1cc80-fdeb-11ea-8e74-94385b63d400.png)

However, I found that when an administrator performs an actual order deletion, the customer analytics data still remains. A fix for this is pending in https://github.com/woocommerce/woocommerce-admin/pull/5171.
